---
title: "Problem set 4"
author: Nova Fong
date: April 1, 2016
output: html_document


---
##Rpubs: http://rpubs.com/fonga153/167279
# Overview

For this problem set you will need to analyze some ChIP-seq data to
identify a mystery factor X.

## Workflow

Create a `run.sh` file that runs the entire workflow (or as much as possible).

### Alignment

Align FASTQ data to the human genome with bowtie2. There are two files
in the `data/` directory:

```
data/factorx.chr1.fq.gz
data/hg19.chr1.fa.gz
```

First build a bowtie2 index with `bowtie2-build` and use `bowtie2` and `samtools` to align the reads to the index.

**The output of the alignment step is a sorted BAM file.**

### Create bedGraph

Create a bedGraph file from the sorted BAM files. Use the
`bedGraphToBigWig` utility and the `hg19.chrom.size` file in the `data/`
directory.

### Create a track in the UCSC browser

1. Create a branch in your forked repository called `gh-pages`:

```bash
$ git branch gh-pages
$ git push origin gh-pages
```

1. Go to the browser and add a "custom track" in the `hg19` genome build.
your trackline should look something like this (all on one line):

```
track type=bedGraph bigDataUrl="http://<username>.github.io/<repo name>/path/to/bw color=255,0,0 visiblity=full name='chip data' description='chip description'
```

### Peak calling

Call peaks from the bedGraph data using MACS2.

```bash
$ macs2 callpeak -t <BAM file>
```


### Generate motifs from the peak calls

1. Use these peak calls to collect FASTA sequences with `bedtools getfasta`.

1. Derive motifs from the FASTA sequences with `meme`.

```bash
# if you get an error about "max size" add -maxsize 1000000
$ meme <FASTA file> -nmotifs 1 -maxw 20 -minw 8 -dna 
```

1. Extract the motif from the `meme.txt` output and use TOMTOM to identify
the motif match. You can use the `meme-get-motif` to extract the first motif from the file:

```bash
meme-get-motif -id 1 < meme.txt
```

Copy the numeric matrix into the the search box on the tomtom site and report which motif it matches.




#Problem Set-4 homework answer:

First i unzipped both of the data files(gunzip hg19.chr1.fa.gz; gunzip factorx.chr1.fq.gz) and then ran my run.sh to identify the factorx.
Below is my "run.sh" file:

#! /usr/bin/env bash
 
datasets='/Users/test/GitHub/ProblemSet1/problem-set-1/problem-set-4/data'

chr1fa="$datasets/hg19.chr1.fa"

factorxchr1fq="$datasets/factorx.chr1.fq"

chromsizes="$datasets/hg19.chrom.sizes"

 
bowtie2-build $chr1fa $chromsizes
 
bowtie2 -x $chromsizes -U $factorxchr1fq | samtools sort > factorx.chr1_sort.bam

bedtools genomecov -ibam factorx.chr1_sort.bam -g $chromsizes -bg > factorx.chr1_sort.bg

bedGraphToBigWig factorx.chr1_sort.bg $chromsizes factorx.chr1_sort.bw

macs2 callpeak -t factorx.chr1_sort.bam -f BAM -n factorx

cat factorx_peaks.narrowPeak | awk 'BEGIN{OFS="\t"} {halfwidth=int(($3-$2)/2); print$1, $2+halfwidth, $2+halfwidth+1}' > factorx_halfwidth.bed

bedtools slop -b 50 -i factorx_halfwidth.bed -g $chromsizes > factorx50bp.bed

bedtools getfasta -fi $chr1fa -bed factorx50bp.bed -fo factorx50bp.fa

###samtools view factorx.chr1_sort.bam | less
###samtools view -H factorx.chr1_sort.bam

meme -minw 8 -maxw 20 -dna -nmotifs 1 factorx50bp.fa -maxsize 10000000

###make url with *.bw file to view onto genome.ucsc.edu website
#track type=bigWig bigDataUrl="http://fonga153.github.io/webstuff/factorx.chr1_sort.bw" color=255,0,0 visibility=full name='factor.chr1_sort.bw' description="factor.chr1"

###took many hours to finish...i started during the morning and didn't finish till the next morning

###i then went to the TOMTOM website and uploaded the meme.txt file from the meme-out folder and got the motif
###matches to NAME:MA0139.1(CTCF)
###Database:JASPAR CORE 2014 vertebrates
###p-value: 6.05e-10
###E-value: 8.67e-07
###q-value: 1,71e-06
###Overlap:19
###Offset: 0
###Orientation:Reverse Complement

###could try another way to identify motif
###go to TOMTOM website and enter in the consensus sequence
###you can use the meme-get-motif to extract the 1st motif from the file

##cp meme.txt meme2.txt
##meme-get-motif -id 1 < meme2.txt
##end of my run.sh 

###Created a track in the UCSC browser by creating a branch in my forked repository called 'gh-pages'.  My custom track looks like below:
track type=bigWig bigDataUrl="http://fonga153.github.io/webstuff/factorx.chr1_sort.bw" color=255,0,0 visibility=full name='factor.chr1_sort.bw' description="factor.chr1"

###Extracted the motif from the 'meme.txt' file on TOMTOM website to identify factorx called CTCF(MA0139.1).  CTCF was the first motif that was identified.



<img src="/Users/test/GitHub/ProblemSet1/problem-set-1/problem-set-4/CTCF_TOMTOMlogo.png" />



